libnetwork/network/non_cni_interface.go

```diff
@@ -0,0 +1,32 @@
+//go:build (linux || freebsd) && !cni
+// +build linux freebsd
+// +build !cni
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/containers/common/libnetwork/types"
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/storage"
+)
+
+const (
+	cniSupported = false
+)
+
+func networkBackendFromStore(store storage.Store, conf *config.Config) (backend types.NetworkBackend, err error) {
+	return types.Netavark, nil
+}
+
+func backendFromType(backend types.NetworkBackend, store storage.Store, conf *config.Config, syslog bool) (types.NetworkBackend, types.ContainerNetwork, error) {
+	if backend != types.Netavark {
+		return "", nil, fmt.Errorf("unsupported network backend %q, check network_backend in containers.conf", backend)
```

Comment:
given we only have two backends  the error feels a bit generic and could mislead users.
I think it is best to just straight up state `podman compiled without cni support` or something like this then it is clear to the user to talk to the pacakger and not file upstream bugs hopefully.